spi/common/core-spi/src/main/java/org/eclipse/edc/spi/iam/IdentityService.java

```diff
@@ -38,8 +38,8 @@ public interface IdentityService {
      * Verifies a JWT bearer token.
      *
      * @param tokenRepresentation A token representation including the token to verify.
-     * @param audience            Expected audience.
+     * @param context             The {@link  VerificationContext}.
      * @return Result of the validation.
      */
-    Result<ClaimToken> verifyJwtToken(TokenRepresentation tokenRepresentation, String audience);
+    Result<ClaimToken> verifyJwtToken(TokenRepresentation tokenRepresentation, VerificationContext context);
```

Comment:
I can already here voices whining about backwards compatibility and breaking changes... Could we leave the old signature in place in a default method, and have it delegate to this new method?
the `VerificationContext` contains the audience anyway

Reply:
We can, but that method will not be called by our code in the protocol service layer, as we would need to populate the `VerificationContext` . Also the `audience` there will go away soon :)

Reply:
@paullatzelsperger hey, that's me :laughing: 
In my opinion it really depends on how many extension of this could exists, seems to be a pretty used interface so I'd go for the deprecation, but I will be also ok in changing without deprecating
(interface changes are generally more tolerated because only devs are involved, instead rest apis... well, you know)